dot/rpc/http_test.go

```diff
@@ -90,3 +100,214 @@ func TestNewHTTPServer(t *testing.T) {
 
 	require.Equal(t, "405 Method Not Allowed", res.Status)
 }
+
+func TestUnsafeRPCProtection(t *testing.T) {
+	cfg := &HTTPServerConfig{
+		Modules: []string{"author"},
+		RPCPort: 7878,
+		RPCAPI:  NewService(),
+	}
+
+	s := NewHTTPServer(cfg)
+	err := s.Start()
+	require.NoError(t, err)
+
+	time.Sleep(time.Second)
+
+	for _, unsafe := range modules.UnsafeMethods {
+		func(method string) {
+			t.Run(fmt.Sprintf("Unsafe method %s should not be reachable", method), func(t *testing.T) {
+				data := []byte(fmt.Sprintf(`{"jsonrpc":"2.0","method":"%s","params":["0x00"],"id":1}`, method))
+
+				buf := new(bytes.Buffer)
+				_, err = buf.Write(data)
+				require.NoError(t, err)
+
+				_, resBody := PostRequest(t, "http://localhost:7878/", buf)
```

Comment:
use `cfg.RPCPort` from above instead of hard coding port `7878`
Change this in other functions as well.

Reply:
Done!